### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by prianka-indla**
Title: HipMRI Study on Prostate Cancer using 2D UNet Model (Easy Difficulty)

Description: This pull request adds a U-Net-based deep learning model for prostate MRI segmentation, including the following components:

Model Architecture (modules.py):
- Implements a U-Net with an encoder-decoder structure and skip connections for accurate segmentation.
- Uses dropout for regularisation and combines Binary Cross-Entropy (BCE) with Dice loss for optimisation.

Data Processing (dataset.py):
- ProstateMRIDataset class loads and preprocesses MRI scans and masks.
- Data augmentation techniques like flips and rotations are applied to improve robustness.

Train and Validate Script (train.py): 
- Initisialise model, load data and train model to accurately predict image segment
- Output 'Dice Score vs Epoch' plot and 'Train and Validate Loss vs Epoch' plot to identify accuracy and efficiency of model

Prediction Script (predict.py):
- Loads the trained model for test inference and visualises results.
- Displays the original image, predicted segmentation, and ground truth, saving the outputs for easy comparison.

ReadMe:
- Includes instructions for environment setup, running the model, and testing.

### CODE REVIEW

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Unclear which dataset was used, validation or training</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Missing references.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Pull Request has problems: Late submission.</td><td>-2</td></tr>
	<tr><td></td><td>Feedback action required: Feedback marks possible +2 if the requested changes are made. Remove cache files for mearge.</td><td>-2</td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-18</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by gayanku** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by shakes76** [comment]
Late submission -10 (2 days). Email me with evidence to restore if you have an extension.

**Comment by shakes76** [comment]
Approved extension +2
